src/windows/CameraProxy.js

```diff
@@ -43,399 +43,468 @@ module.exports = {
+                                    .then(function() {
+                                        // Let Windows know that we're finished changing the file so
+                                        // the other app can update the remote version of the file.
+                                        return Windows.Storage.CachedFileManager.completeUpdatesAsync(file);
+                                    })
+                                    .done(function(updateStatus) {
+                                            if (updateStatus === Windows.Storage.Provider.FileUpdateStatus.complete) {
+                                                success(capturedFile);
+                                            } else {
+                                                errorCallback();
+                                            }
+                                    }, errorCallback);
                             } else {
-                                success(capturedFile);
+                                errorCallback();
                             }
+                            Windows.UI.WebUI.WebUIApplication.removeEventListener("activated", cameraActivationHandler);
+                        }
+                    };
 
-
-                        }, function(err) {
-                            destroyCameraPreview();
-                            errorCallback(err);
-                        });
-                    }, function(err) {
-                        destroyCameraPreview();
-                        errorCallback(err);
-                    });
-                };
-
-                try {
-                    createCameraUI();
-                    startCameraPreview();
-                } catch (ex) {
-                    errorCallback(ex);
+                    var savePicker = new Windows.Storage.Pickers.FileSavePicker();
+                    savePicker.suggestedStartLocation = Windows.Storage.Pickers.PickerLocationId.documentsLibrary;
+                    if (encodingType === Camera.EncodingType.PNG) {
+                        savePicker.fileTypeChoices.insert("PNG", [".png"]);
+                    } else {
+                        savePicker.fileTypeChoices.insert("JPEG", [".jpg"]);
+                    }
+                    savePicker.suggestedFileName = fileName;
+                    Windows.UI.WebUI.WebUIApplication.addEventListener("activated", cameraActivationHandler);
+                    savePicker.pickSaveFileAndContinue();
+                } else {
+                    success(capturedFile);
                 }
+            }, function(err) {
+                    destroyCameraPreview();
+                    errorCallback(err);
+            });
+    };
 
-            } else {
+    try {
+        createCameraUI();
+        startCameraPreview();
+    } catch (ex) {
+        errorCallback(ex);
+    }
+}
+
+function takePictureFromCameraWindows(successCallback, errorCallback, args) {
+    var destinationType = args[1],
```

Comment:
nit: There is exactly the same code exist in `takePictureFromCameraWP`. You may want to move it to upper-level `takePictureFromCamera`, construct an options object and pass it downstream.